pandas/io/stata.py

```diff
@@ -466,6 +466,12 @@ def parse_dates_safe(dates, delta=False, year=False, days=False):
 """
 
 
+general_object_array_error = """
+Writing non-string object columns is not supported. Column '%s' does not
```

Comment:
We've been moving to new-style format strings. Could you change the `%s` to `{}`?

Also for relatively simply error message like this, having the message next to where it's raised helps keep the code clearer. Could you define this message where it's used? Use params to wrap lines in the source

```
msg = (
    "Writing non-string... "
    " ... "
)
raise ValueError(msg)
```